plugins/scaffolder-backend/src/scaffolder/stages/publish/types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import { RequiredTemplateValues } from '../templater';
+import { JsonValue } from '@backstage/config';
+
+export type Publisher = {
+  publish(opts: {
+    entity: TemplateEntityV1alpha1;
+    values: RequiredTemplateValues & Record<string, JsonValue>;
+    directory: string;
+  }): Promise<{ remoteUrl: string }>;
```

Comment:
Is the `remoteUrl` actually used right now? Otherwise I'd say just return void for now.

Reply:
Yep, it's used to register the entire entry into the catalog. The frontend needs it, as the there's no service discovery for the scaffolder to talk to the catalog, so the frontend registers with the catalog.

Thinking in the log term though, I really think we need to have a plan for service discovery and services talking to each other. I get the benefits of things being standalone. Services just being input and output, but we can really in no way rely on the frontend to keep consistency between calls to multiple services.

Reply:
Yep, service discovery needs to be a thing.

Regarding the return value, what do you think about making it a Location entity instead? Thinking that re-uses the definitions for different types of locations, and is likely very forwards compatible. The frontend would then just grab the target and post to the catalog for now. Not thinking we'd add it to this PR though.